flink-runtime/src/test/java/org/apache/flink/runtime/io/network/partition/SortMergeResultPartitionReadSchedulerTest.java

```diff
@@ -192,6 +194,53 @@ public void testOnReadBufferRequestError() throws Exception {
         assertAllResourcesReleased();
     }
 
+    @Test
+    public void testIncreaseDeadlineWhenRecycleBuffer() throws Exception {
+        testRequestDeadlineInternal(true);
+        assertTrue(readScheduler.getBufferRequestDeadline().hasTimeLeft());
```

Comment:
I guess we can make allocateBuffers package private (visible for testing) and verify the timeout exception. Only verify the deadline without verify the end result seems a little weak. BTW, in this test, verifying the deadline still has left may make the test unstable in the slow test environment.

Reply:
Ok, I have updated the tests.